src/test/java/itest/RulesPostMultipartIT.java

```diff
@@ -0,0 +1,140 @@
+                        ar -> {
+                            if (ar.succeeded()) {
+                                deleteResponse.complete(ar.result().bodyAsJsonObject());
+                            } else {
+                                deleteResponse.completeExceptionally(ar.cause());
+                            }
+                        });
+
+        JsonObject expectedDeleteResponse =
+                new JsonObject(
+                        Map.of(
+                                "meta",
+                                Map.of("type", HttpMimeType.JSON.mime(), "status", "OK"),
+                                "data",
+                                NULL_RESULT));
+
+        try {
+            JsonObject deleteResult = deleteResponse.get(5, TimeUnit.SECONDS);
+            MatcherAssert.assertThat(deleteResult, Matchers.equalTo(expectedDeleteResponse));
+        } catch (Exception e) {
+            System.out.println("Deletion failed. Reason: " + e.getMessage());
+            throw e;
+        }
+    }
+
+    @ParameterizedTest
+    @ValueSource(
+            strings = {
+                "multipart/form-data",
+                "multipart/form-data; boundary=------somecharacters",
+                "multipart/form-data; unkown characters",
+                "multipart/form-data; directive1; directive2",
+                "multipart/form-data; directive"
+            })
+    void shouldAcceptMultipartWithBoundary(String contentType) throws Exception {
+        MultiMap form = MultiMap.caseInsensitiveMultiMap();
+        form.set("name", TEST_RULE_NAME);
+        form.set("matchExpression", "false");
+        form.set("eventSpecifier", "template=Continuous");
+
+        CompletableFuture<JsonObject> response = new CompletableFuture<>();
+        webClient
+                .post("/api/v2/rules")
+                .putHeader(HttpHeaders.CONTENT_TYPE.toString(), contentType)
+                .sendForm(
+                        form,
+                        ar -> {
+                            if (assertRequestStatus(ar, response)) {
+                                response.complete(ar.result().bodyAsJsonObject());
+                            } else {
+                                response.completeExceptionally(
+                                        new RuntimeException("Request failed"));
+                            }
+                        });
+        try {
+            JsonObject result = response.get(5, TimeUnit.SECONDS);
+            System.out.println("Received response: " + result.toString());
+
+            MatcherAssert.assertThat(result.getJsonObject("meta"), Matchers.notNullValue());
+            MatcherAssert.assertThat(
+                    result.getJsonObject("meta").getString("status"), Matchers.equalTo("Created"));
+            MatcherAssert.assertThat(result.getJsonObject("data"), Matchers.notNullValue());
+        } catch (Exception e) {
+            System.err.println("Request failed: " + e.getMessage());
```

Comment:
Same question here about catching/logging/rethrowing as I had before about the `TimeoutException`. Why not just let the exception be thrown?

Reply:
you mean I should get rid of the try?

Reply:
Yep, pretty much.